core/domain/action_registry_test.py

```diff
@@ -29,3 +29,12 @@ def test_action_registry(self):
         """Do some sanity checks on the action registry."""
         self.assertEqual(
             len(action_registry.Registry.get_all_actions()), 3)
+
+    def test_get_action_by_type(self):
+        # Testing with invalid action type.
+        with self.assertRaisesRegex(KeyError, 'fakeAction'):
```

Comment:
Can you make the second argument more specific to the error message we expect to see?

Reply:
No error message was explicitly defined for KeyError in action_registry.py. So I replaced it with assertRaises(KeyError) rather than using assertRaisesRegex( ) to check for an error message.

Reply:
The KeyError comes from this line:

https://github.com/oppia/oppia/blob/7339ac6fa6b80b6bee87faf05ea2dc8e3dd999c1/core/domain/action_registry.py#L91

You can see the form of this error like this:

```python
>>> d = {1: 2}
>>> d[3]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 3
```

So I think you can use `assertRaisesRegex` with a message of the form `KeyError: fakeAction`

Reply:
Thanks for the review @U8NWXD. I tried using `assertRaisesRegex(KeyError, 'KeyError: fakeAction')` but it still checks for the invalid key name - 
![Screenshot from 2022-01-11 09-05-14](https://user-images.githubusercontent.com/89375125/148877035-c860938b-4de7-48fe-8ee2-b352de1724b6.png)

Reply:
I think you need to add quotes: `KeyError: 'fakeAction'`

Reply:
Thanks @U8NWXD. It returns the invalid key_name as the error message and `KeyError: 'fakeAction'` doesn't seem to work. Error log - 
![Screenshot from 2022-01-12 09-40-47](https://user-images.githubusercontent.com/89375125/149062659-d1c890b5-0ea4-4b66-8221-705399e18396.png)

Reply:
OK, then it looks like the error message is just `'fakeAction'`. I think that means you were right initially. Maybe just add a comment explaining this